apps/condo/domains/meter/schema/RegisterMetersReadingsService.js

```diff
@@ -0,0 +1,260 @@
+                        organization,
+                        property: { id: property.id },
+                        unitType: reading.addressMeta.unitType,
+                        unitName: reading.addressMeta.unitName,
+                        accountNumber: reading.accountNumber,
+                        number: reading.meterNumber,
+                    })
+
+                    if (foundMeter) {
+                        meterId = foundMeter.id
+                    } else {
+                        try {
+                            const createdMeter = await Meter.create(context, {
+                                dv,
+                                sender,
+                                organization: { connect: organization },
+                                property: { connect: { id: property.id } },
+                                unitType: reading.addressMeta.unitType,
+                                unitName: reading.addressMeta.unitName,
+                                accountNumber: reading.accountNumber,
+                                number: reading.meterNumber,
+                                resource: { connect: reading.meterResource },
+                                numberOfTariffs: reading.meterMeta.numberOfTariffs,
+                                place: reading.meterMeta.place,
+                                verificationDate: toISO(reading.meterMeta.verificationDate),
+                                nextVerificationDate: toISO(reading.meterMeta.nextVerificationDate),
+                                installationDate: toISO(reading.meterMeta.installationDate),
+                                commissioningDate: toISO(reading.meterMeta.commissioningDate),
+                                sealingDate: toISO(reading.meterMeta.sealingDate),
+                                controlReadingsDate: reading.meterMeta.controlReadingsDate ? toISO(reading.meterMeta.controlReadingsDate) : dayjs().toISOString(),
+                            })
+                            meterId = createdMeter.id
+                        } catch (e) {
+                            resultRows.push(e)
+                            continue
+                        }
+                    }
+
+                    try {
+                        const values = ['value1', 'value2', 'value3', 'value4'].reduce((result, currentValue) => {
+                            const value = reading[currentValue]
+                            const normalizedValue = normalizeMeterValue(value)
+
+                            if (!validateMeterValue(normalizedValue)) {
+                                throw new GQLError(ERRORS.INVALID_METER_VALUE(currentValue, value), context)
+                            }
+
+                            return {
+                                ...result,
+                                [currentValue]: normalizedValue,
+                            }
+                        }, {})
```

Comment:
can we check all the values at once to return all errors, not just the first one?

Reply:
It looks like it is not necessary. We check, for example, property in the same manner. We do not need to collect all errors for the particular row.

Reply:
Ok

Just my fantasies :)
For me as a user, it would be more convenient to receive all the errors at once rather than one by one.
If I'm not mistaken, this is how the current import works

Reply:
fixed :)